cmd/kubeadm/app/cmd/init.go

```diff
@@ -332,21 +332,40 @@ func newInitData(cmd *cobra.Command, args []string, options *initOptions, out io
 		}
 	}
 
-	// Checks if an external CA is provided by the user.
-	externalCA, _ := certsphase.UsingExternalCA(&cfg.ClusterConfiguration)
+	// Checks if an external CA is provided by the user (when the CA Cert is present but the CA Key is not)
+	externalCA, err := certsphase.UsingExternalCA(&cfg.ClusterConfiguration)
```

Comment:
The error handling here is a bit convoluted with the internal logic of `UsingExternalCA`, when no externalCA this error is never checked and we can't tell the difference between an unrelated error (cannot read a file because of permission errors for example) from errors that are related to x509 validations.

Reply:
yep that's true, but at least the error message that surfaces to the user can help to understand and fix the problem

ATM UsingExternalCA returns
- false, with error in case there is no CA
- false, nil in case  in case the user provided a CA but it is not external
- true, with error in case the user provided an external CA but the signed certs are not valid
- true, nil in case the user provided an external CA and the signed certs are valid

After release this should be probably refactored